src/sync/channel.zig

```diff
@@ -125,6 +125,27 @@ pub fn Channel(comptime T: type) type {
             return out;
         }
 
+        pub fn tryDrainRecycle(
+            self: *Self,
+            buf: *std.ArrayList(T),
+        ) error{ ChannelClosed, OutOfMemory }!void {
+            var buffer = self.buffer.lock();
+            defer buffer.unlock();
+            buf.clearRetainingCapacity();
```

Comment:
I guess my thought process is that once you hand over a buf to a function that's going to fill the buf, you should expect the buf to get overwritten. Would you expect something different because this is an arraylist?

Reply:
yeah i think the arraylist is the part that confuses me - but nbd